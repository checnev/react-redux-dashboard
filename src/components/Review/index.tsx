import React from 'react';
import {
  Grid,
  Typography,
  Divider,
  Paper,
} from '@material-ui/core';
import { Rating } from '@material-ui/lab';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import type { Moment } from 'moment';

export interface IReview {
  id: number;
  author: string;
  date: Moment;
  rating: number;
  text: string;
}

const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {},
  paper: {
    padding: theme.spacing(2),
  },
  header: {},
  author: {},
  date: {},
  divider: {
    marginTop: -12,
    marginBottom: -12,
  }
}));

const Review: React.FC<IReview> = ({ id, author, date, rating, text } : IReview) => {
  const classes = useStyles();

  return (
    <Paper className={classes.paper}>
      <Grid container className={classes.root} spacing={3}>
        <Grid item xs={12} sm={6}>
          <Typography className={classes.author} variant="h5" component="h4">{author}</Typography>
          <div>
            <Rating name="rating" value={rating} precision={0.5} readOnly size="small" />
          </div>
        </Grid>
        <Grid item xs={12} sm={6}>
          <Typography
            className={classes.date}
            color="textSecondary"
            align="right"
          >
            {date.format('DD.MM.YYYY')}
          </Typography>
        </Grid>
        <Grid item xs={12} className={classes.divider}><Divider /></Grid>
        <Grid item xs={12}>
          <Typography color="textSecondary" dangerouslySetInnerHTML={{ __html: text}} />
        </Grid>
      </Grid>
    </Paper>
  );
}

export default Review;
