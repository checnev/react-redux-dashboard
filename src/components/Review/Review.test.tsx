import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Review, { IReview } from 'components/Review';
import moment from 'moment';

const review: IReview = {
  id: 1,
  author: 'Satoshi Nakamuto',
  date: moment('10.12.2020', 'DD.MM.YYYY'),
  rating: 4.5,
  text: 'test text',
};

let container: any = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Review render', () => {
  act(() => {
    render(<Review {...review} />, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
