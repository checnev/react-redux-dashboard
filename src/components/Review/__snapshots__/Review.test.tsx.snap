// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review render 1`] = `
"<div class=\\"MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded\\">
  <div class=\\"MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-3\\">
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\">
      <h4 class=\\"MuiTypography-root makeStyles-author-4 MuiTypography-h5\\">Satoshi Nakamuto</h4>
      <div><span class=\\"MuiRating-root MuiRating-sizeSmall MuiRating-readOnly\\" role=\\"img\\" aria-label=\\"4.5 Stars\\"><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 50%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconEmpty\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span></span></div>
    </div>
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\">
      <p class=\\"MuiTypography-root makeStyles-date-5 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignRight\\">10.12.2020</p>
    </div>
    <div class=\\"MuiGrid-root makeStyles-divider-6 MuiGrid-item MuiGrid-grid-xs-12\\">
      <hr class=\\"MuiDivider-root\\">
    </div>
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\">
      <p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">test text</p>
    </div>
  </div>
</div>"
`;
